pkg/webhook/addons/registry/autoenabler.go

```diff
@@ -0,0 +1,185 @@
+// Copyright 2025 Nutanix. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package registry
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	ctrlclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	carenv1 "github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/api/v1alpha1"
+	"github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/api/variables"
+	capiutils "github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/common/pkg/capi/utils"
+)
+
+type workloadClusterAutoEnabler struct {
+	client  ctrlclient.Client
+	decoder admission.Decoder
+}
+
+func NewWorkloadClusterAutoEnabler(
+	client ctrlclient.Client, decoder admission.Decoder,
+) *workloadClusterAutoEnabler {
+	return &workloadClusterAutoEnabler{
+		client:  client,
+		decoder: decoder,
+	}
+}
+
+func (a *workloadClusterAutoEnabler) Defaulter() admission.HandlerFunc {
+	return a.defaulter
+}
+
+func (a *workloadClusterAutoEnabler) defaulter(
+	ctx context.Context,
+	req admission.Request,
+) admission.Response {
+	cluster := &clusterv1.Cluster{}
+	err := a.decoder.Decode(req, cluster)
+	if err != nil {
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+
+	if hasSkipAnnotation(cluster) {
+		return admission.Allowed("")
+	}
+
+	if cluster.Spec.Topology == nil {
+		return admission.Allowed("")
+	}
+
+	// Check if the addon is already enabled in the cluster, if it is just return.
+	clusterRegistry, err := registryAddonFromCluster(cluster)
+	if err != nil {
+		return admission.Errored(
+			http.StatusInternalServerError,
+			fmt.Errorf(
+				"failed to check if registry addon is enabled in cluster: %w",
+				err,
+			))
+	}
+	if clusterRegistry != nil {
```

Comment:
not directly related to this default webhook. So if the management cluster does not have registry addon enabled, but workload cluster has it set, we will deploy in-cluster registry on the workload cluster. If the other flag to sync images is not enabled then this registry on workload cluster will not be used. 
do we want to allow such usecase? 
Or separately we should block cluster creation in above case?

Reply:
>If the other flag to sync images is not enabled then this registry on workload cluster will not be used.

The other webhook will also check if its enabled on the mgmt cluster and only set if it is.

> do we want to allow such usecase?

Yes I think we should, its possible to seed the registry in other ways.